frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -83,16 +84,38 @@ export const useFacilityValidation = (facility: Facility) => {
         {} as FacilityErrors
       );
       setErrors(errors);
+      focusOnce.current = true;
       showError(
         "Please check the form to make sure you complete all of the required fields.",
         "Form Errors"
       );
-      let firstError = document.querySelector("[aria-invalid=true]");
-      (firstError as HTMLElement)?.focus();
       return "error";
     }
   };
 
+  /**
+   * Focus on fields with errors
+   */
+  useEffect(() => {
+    if (
+      focusOnce.current &&
+      (errors.name ||
+        errors.phone ||
+        errors.street ||
+        errors.zipCode ||
+        errors.state ||
+        errors.cliaNumber ||
+        errors["orderingProvider.firstName"] ||
+        errors["orderingProvider.lastName"] ||
+        errors["orderingProvider.NPI"] ||
+        errors["orderingProvider.phone"])
```

Comment:
I think we also need `errors.deviceTypes` here in case a user gets rid of all the devices in their facility and clicks the save button at the top of the form

Reply:
Thanks for pointing this out! I was hoping someone more familiar with the logic would chime in with suggestions :D